File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.js
/**
 * @param {number[][]} edges
 * @return {number}
 */
var findCenter = function(edges) {
    const graph = {};
    
    for([s, e] of edges) {
        if(!graph[s]) graph[s] = []
        graph[s].push(e)
        
        if(!graph[e]) graph[e] = []
        graph[e].push(s)
    }
    
    for(const node in graph){
        if(graph[node].length === edges.length) return node;
    }
    
    return 1
};

File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int findCenter(int[][] edges) {
        int n = edges.length + 1;
        List<Integer>[] graph = new ArrayList[n+1];
        
        for(int i = 1; i <= n; i++){
            graph[i]  = new ArrayList<>();
        }
        
        for(int[] edge: edges){
            int s = edge[0];
            int e = edge[1];
            graph[s].add(e);
            graph[e].add(s);
        }
        
        for(int i = 1; i <= n; i++) {
            if(graph[i].size() == edges.length) return i;
        }
        
        return -1;
    }
}